x/bank/keeper/msg_server.go

```diff
@@ -2,19 +2,30 @@ package keeper
 
 import (
 	"context"
-
 	"github.com/armon/go-metrics"
-
+	"cosmossdk.io/errors"
 	"github.com/cosmos/cosmos-sdk/telemetry"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/x/bank/types"
+	gov "github.com/cosmos/cosmos-sdk/x/gov/types"
 )
 
 type msgServer struct {
 	Keeper
 }
 
+// UpdateDenomMetadata updates the denom metadata if the message is signed by gov module account
+func (k msgServer) UpdateDenomMetadata(goCtx context.Context, msg *types.MsgUpdateDenomMetadata) (*types.MsgUpdateDenomMetadataResponse, error) {
```

Comment:
I would like to see some unit tests on this function:
1. Providing the wrong authority returns the correct error.
2. Providing the correct authority causes the denom metadata to be updated as provided.

I realize there aren't any unit tests on the bank module's msg_server funcs, but there really should be. I guess there's a few tests on the CLI commands which would pass things through here, so that's probably why there isn't a `msg_server_test.go`.

Speaking of which, there should be a command for this too. That being said, it's probably good to do that as a separate PR anyway since, judging by what's in the denom metadata, that command is probably going to have lots of arguments/flags. Also, I'm not exactly sure how that command should behave since it's for a gov proposal. The only example I've looked at is for a software upgrade, but I don't know if they've changed their minds on stuff since then (I'd have to look at upstream `main`).